doc/basics.md

```diff
@@ -2621,6 +2623,120 @@ Performance tips
   ```
 - To better understand the operation of your On Demand parser, and whether it is performing as well as you think it should be, there is a  logger feature built in to simdjson! To use it, define the pre-processor directive `SIMDJSON_VERBOSE_LOGGING` prior to including the `simdjson.h` header, which enables logging in simdjson. Run your code. It may generate a lot of logging output; adding printouts from your application that show each section may be helpful. The log's output will show step-by-step information on state, buffer pointer position, depth, and key retrieval status. Importantly, unless `SIMDJSON_VERBOSE_LOGGING` is defined, logging is entirely disabled and thus carries no overhead.
 
+
+Advanced Topics: Free Padding
+-------
+
+For performance reasons, the simdjson library requires that the JSON input contain at least
+`simdjson::SIMDJSON_PADDING` bytes at the end of the stream. The value `simdjson::SIMDJSON_PADDING` is
+small (e.g., 64 bytes). On modern systems, you can safely read beyond an allocated buffers,
+as long as you remain within an allocated page. Pages on modern systems span at least 4 kilobytes,
+but can be significantly larger. E.g., Apple systems favour pages spanning 16 kilobytes.
+
+In effect, it means that you can almost always read a few bytes beyond your current buffer.
+Thus, even though you may sometimes need to reallocate the memory to get extra padding, this should be
+uncommon. However, tools such as valgrind or memory sanitizers will flag such behavior as unsafe.
+While the early releases of the simdjson made use of this capability, we removed it due to avoid
```

Comment:
I think we got rid of this behavior [in stage 1](https://github.com/simdjson/simdjson/blob/master/src/generic/stage1/buf_block_reader.h#L101-L102), but we still read into the padding sometimes (in [string unescaping](https://github.com/simdjson/simdjson/blob/master/src/generic/stage2/stringparsing.h#L152-L153) at the least).

Reply:
I'll trim this comment since it seems unclear.